test/system/src/test/java/io/pravega/test/system/ConsumptionBasedRetentionWithMultipleReaderGroupsTest.java

```diff
@@ -391,6 +409,119 @@ public void updateRetentionPolicyForCBRTest() throws Exception {
+
+        // Write two more events.
+        writingEventsToStream(2, writer, scope, stream);
+
+        // Retention set has one stream cut at 0/150
+        // READER_GROUP_1 updated stream cut at 0/30
+        // Subscriber lower bound is 0/30, truncation should happen at this point
+        // The timeout is set to 2 minutes a little longer than the retention period which is set to 1 minutes
+        // in order to confirm that the retention has taken place.
+        AssertExtensions.assertEventuallyEquals("Truncation did not take place at offset 30.", true, () -> controller.getSegmentsAtTime(
+                        new StreamImpl(scope, stream), 0L).join().values().stream().anyMatch(off -> off == 30),
+                5000, 2 * 60 * 1000L);
+
+        // Validating the failover scenario - Start
+        log.info("Controller and segment store failover scenario started");
+        // Write six events.
+        writingEventsToStream(3, writer, scope, stream);
+        // Read two events with reader.
+        readingEventsFromStream(1, reader);
+
+        checkpoint = initiateCheckPoint("Checkpoint3", readerGroup, reader, clock);
+        read = reader.readNextEvent(READ_TIMEOUT);
+        log.info("Reading next event after checkpoint3 {}", read.getEvent());
+        cpResult = checkpoint.join();
+        assertTrue(checkpoint.isDone());
+
+        //Controller Failover
+        scaleAndUpdateControllerURI(1);
+        log.info("Successfully scaled down controller to 1 instance");
+        //SegmentStore Failover
+        Futures.getAndHandleExceptions(segmentStoreService.scaleService(1), ExecutionException::new);
+        log.info("Successfully scaled down segment store to 1 instance");
+
+        // Retention set has two stream cut at 0/150...0/240
+        // READER_GROUP_1 updated stream cut at 0/90
+        // Subscriber lower bound is 0/90, truncation should happen at this point
+        AssertExtensions.assertEventuallyEquals("Truncation did not take place at offset 90.", true, () -> controller.getSegmentsAtTime(
+                        new StreamImpl(scope, stream), 0L).join().values().stream().anyMatch(off -> off == 90),
+                5000,  2 * 60 * 1000L);
+
+        // Validating the restart scenario - Start
+        checkpoint = initiateCheckPoint("Checkpoint4", readerGroup, reader, clock);
+        read = reader.readNextEvent(READ_TIMEOUT);
+        log.info("Reading next event after checkpoint4 {}", read.getEvent());
+        cpResult = checkpoint.join();
+        assertTrue(checkpoint.isDone());
+
+        //Closing all the resources before restart
+        reader.close();
+        readerGroup.close();
+        readerGroupManager.close();
+        writer.close();
+        clientFactory.close();
+        connectionFactory.close();
+        Futures.getAndHandleExceptions(controllerService.scaleService(0), ExecutionException::new);
+        log.info("Ankur Successfully stopped 1 instance of controller service");
```

Comment:
Its not recommended to have developer's name in log.

Reply:
@AnkurAnand11 , if you feel that your PR is not ready then change it to draft mode. Clean up and check your PR from your side first then only open it up for review